x-pack/plugins/fleet/server/services/agent_policy.ts

```diff
@@ -737,24 +749,41 @@ class AgentPolicyService {
           }),
     };
 
+    const permissions = Object.fromEntries(
+      await Promise.all(
+        // Original type is `string[] | PackagePolicy[]`, but TS doesn't allow to `map()` over that.
```

Comment:
I'm not sure if `string[]` still applies, but it's on the type definition for `agentPolicy.package_policies` so I'm handling that case for now.